torch/lib/THD/base/init_methods/InitMethodTCP.cpp

```diff
@@ -35,8 +35,11 @@ std::string getRandomString()
 
   int fd;
   unsigned int seed;
+  ssize_t bytes_read;
   SYSCHECK(fd = open("/dev/urandom", O_RDONLY));
-  SYSCHECK(read(fd, &seed, sizeof(seed)));
+  SYSCHECK(bytes_read = read(fd, &seed, sizeof(seed)));
```

Comment:
Because C++11 random library either doesn't guarantee anything about the seed, or uses a fixed one, while we need to somehow break symmetry between machines.